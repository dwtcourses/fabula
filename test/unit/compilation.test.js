
import { readFileSync } from 'fs'
import { compileCommand } from '../util'

describe('task compilation', () => {

  test('create-setup.sh', () => {

    const templateFile = readFileSync('test/fixtures/setup-ssh.sh')

    const sampleSettings = {
      hostname: 'my-server',
      host: '192.168.100.100',
      port: 22,
      username: 'username',
      privateKey: '/here/is/my/key'
    }

    const compiled = compileCommand(templateFile, sampleSettings)
    expect(JSON.stringify(compiled, null, 2)).toMatchSnapshot()
  })
})
