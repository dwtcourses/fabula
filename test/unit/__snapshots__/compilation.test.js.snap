// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`task compilation create-setup.sh 1`] = `
"[
  \\"rm -rf /instance $HOME/.keys $HOME/.ssh/config\\",
  \\"mkdir -p /instance/ $HOME/.keys\\",
  \\"chown -R ubuntu /instance/ $HOME/.keys\\",
  \\"chmod 755 $HOME/.keys\\",
  [
    \\"put\\",
    [
      \\"/my/key $HOME/.key\\",
      \\"/deploy_key\\"
    ]
  ],
  \\"chmod 400 $HOME/.keys/deploy_key\\",
  [
    \\"echo\\",
    \\"$HOME/.ssh/config\\",
    [
      \\"  Host 192.168.100.100\\",
      \\"    Hostname my-server\\",
      \\"    Port 22\\",
      \\"    IdentityFile $HOME/.keys/deploy_key\\",
      \\"    User username\\",
      \\"    StrictHostKeyChecking no\\"
    ]
  ]
]"
`;
